django_project/feti/management/commands/scraping_occupation.py

```diff
@@ -61,153 +158,193 @@ def create_occupation(data):
     LearningPathway.objects.filter(occupation=occupation).delete()
     pathways = []
     for learning_pathway in data['learning_pathways']:
+        print('Create learning pathway...')
         pathway = create_learning_pathway(learning_pathway, occupation)
         for step in learning_pathway['steps']:
+            print('Create step...')
             create_step(step, pathway)
         pathways.append(pathway)
     return occupation
 
 
+def scraping_occupations(html):
+    items = html.findAll("div", {"class": "SearchResultItem"})
+    if len(items) == 0:
+        print('this page is empty')
+        return False
+    else:
+        is_empty = False
+        # tracing each of item
+        for item in items:
+            print('--------------------------------')
+            if item.string and item.string == "No Records Found.":
+                is_empty = True
+                print('No records found')
+                break
+            # get occupation information
+            if item.find("a", {"class": "linkNoMore"}):
+                pass
+            else:
+                print('Parsing the occupation list html...')
+                occupation = dict()
+                occupation['occupation'] = cleaning(item.a.string.split("Occupation Code")[0][:-1])
+                occupation['green_occupation'] = False
+                occupation['scarce_skill'] = False
+                # checking green
+                if item.find("span", {"class": "greenTagListView"}):
+                    occupation['green_occupation'] = True
+                if item.find("span", {"class": "scarceTagListView"}):
+                    occupation['scarce_skill'] = True
+
+                # get detail
+                print('Open learning pathway...')
+                html = get_soup(item.a.get('href'))
+                body = html.find("div", {"class": "BodyPanel642"})
+                if body:
+                    print('Parsing learning pathway...')
+                    content = str(body)
+                    content = content.split("<b>Description</b>")
+                    if len(content) > 1:
+                        content = content[1]
+                        # description
+                        occupation['description'] = cleaning(beautify(content.split("<b>")[0]).get_text())
+
+                        # tasks
+                        splits = content.split("<b>Tasks</b>")
+                        occupation['tasks'] = beautify(splits[1].split("<b>")[0]).get_text() \
+                            if len(splits) > 1 else ""
```

Comment:
when we need to  split single line if's over more than 1 line, its better to keep them as traditional if: else: statements.